src/plugins/orangehrmClaimPlugin/Api/ClaimExpenseTypeAPI.php

```diff
@@ -41,25 +46,101 @@ class ClaimExpenseTypeAPI extends Endpoint implements CrudEndpoint
+     *         @OA\Schema(type="string", enum=ClaimExpenseTypeSearchFilterParams::ALLOWED_SORT_FIELDS)
+     *     ),
+     *     @OA\Parameter(ref="#/components/parameters/sortOrder"),
+     *     @OA\Parameter(ref="#/components/parameters/limit"),
+     *     @OA\Parameter(ref="#/components/parameters/offset"),
+     *     @OA\Response(
+     *         response="200",
+     *         description="Success",
+     *         @OA\JsonContent(
+     *             @OA\Property(
+     *                 property="data",
+     *                 type="array",
+     *                 @OA\Items(ref="#/components/schemas/Claim-ClaimExpenseTypeModel")
+     *             ),
+     *             @OA\Property(property="meta",
+     *                 type="object",
+     *                 @OA\Property(property="total", type="integer")
+     *             )
+     *         )
+     *     )
+     * )
      * @inheritDoc
      */
     public function getAll(): EndpointResult
     {
-        throw $this->getNotImplementedException();
+        $claimExpenseTypeSearchFilterParams = new ClaimExpenseTypeSearchFilterParams();
+        $this->setSortingAndPaginationParams($claimExpenseTypeSearchFilterParams);
+        $claimExpenseTypeSearchFilterParams->setName(
+            $this->getRequestParams()->getStringOrNull(RequestParams::PARAM_TYPE_QUERY, self::PARAMETER_NAME)
+        );
+        $claimExpenseTypeSearchFilterParams->setStatus(
+            $this->getRequestParams()->getBooleanOrNull(RequestParams::PARAM_TYPE_QUERY, self::PARAMETER_STATUS)
+        );
+        $claimExpenseTypeSearchFilterParams->setId(
+            $this->getRequestParams()->getIntOrNull(RequestParams::PARAM_TYPE_QUERY, self::PARAMETER_EXPENSE_TYPE_ID)
+        );
+        $claimExpenseTypes = $this->getClaimService()->getClaimDao()->getExpenseTypeList(
+            $claimExpenseTypeSearchFilterParams
+        );
+        $count = $this->getClaimService()->getClaimDao()->getClaimExpenseTypeCount($claimExpenseTypeSearchFilterParams);
+        return new EndpointCollectionResult(
+            ClaimExpenseTypeModel::class,
+            $claimExpenseTypes,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => $count])
+        );
     }
 
     /**
      * @inheritDoc
+     * @return ParamRuleCollection
      */
     public function getValidationRuleForGetAll(): ParamRuleCollection
     {
-        throw $this->getNotImplementedException();
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_NAME,
+                    new Rule(
+                        Rules::STRING_TYPE
+                    )
+                )
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_STATUS,
+                    new Rule(
+                        Rules::STRING_TYPE
```

Comment:
```suggestion
                        Rules::BOOL_VAL
```
is this boolean or string? :thinking:

Reply:
i kept it as String, unless it fail my test cases.